compose/upgrade.md

```diff
@@ -1,184 +1,133 @@
+It offers an improved CLI experience, improved build performance with BuildKit, and continued new feature development.
 
-<hr>
-</div>
-<div id="tab2" class="tab-pane fade" markdown="1">
+For most projects, switching to Compose V2 requires no changes to the Compose YAML or your development workflow.
 
-The list below includes the flags that we are not planning to support in Compose in the Docker CLI,
-either because they are already deprecated in `docker-compose`, or because they are not relevant for Compose in the Docker CLI.
+### `docker-compose` vs `docker compose`
 
-* `compose ps --filter KEY-VALUE` Not relevant due to its complicated usage with the `service` command and also because it is not documented properly in `docker-compose`.
-* `compose rm --all` Deprecated in docker-compose.
-* `compose scale` Deprecated in docker-compose (use `compose up --scale` instead)
+Unlike Compose V1, Compose V2 integrates into the Docker CLI platform and the recommended command-line syntax is `docker compose`.
 
-Global flags:
+The Docker CLI platform provides a consistent and predictable set of options and flags, such as the `DOCKER_HOST` environment variable or the `--context` command-line flag.
 
-* `--compatibility` has been resignified Docker Compose V2. This now means that in the command running V2 will behave as V1 used to do.
-  * One difference is in the word separator on container names. V1 used to use `_` as separator while V2 uses `-` to keep the names more hostname friendly. So when using `--compatibility` Docker 
-    Compose should use `_` again. Just make sure to stick to one of them otherwise Docker Compose will not be able to recognize the container as an instance of the service.
-<hr>
-</div>
-<div id="tab3" class="tab-pane fade" markdown="1">
+This change enables using all of the shared flags on the root `docker` command.
+For example, `docker --log-level=debug --tls compose up` enables debug logging from the Docker Engine as well as ensures that TLS is used for the connection.
 
-#### Copy
+Update scripts to use Compose V2 by replacing the hyphen (`-`) with a space, using `docker compose` instead of `docker-compose`.
 
-The `cp` command is intended to copy files or folders between service containers and the local filesystem.  
-This command is a bidirectional command, we can copy **from** or **to** the service containers.
+### Service container names
 
-Copy a file from a service container to the local filesystem:
+Compose generates container names based on the project name, service name, and scale/replica count.
 
-```console
-$ docker compose cp my-service:~/path/to/myfile ~/local/path/to/copied/file
-```
+In Compose V1, an underscore (`_`) was used as the word separator.
+In Compose V2, a hyphen (`-`) is used as the word separator.
 
-We can also copy from the local filesystem to all the running containers of a service:
+Underscores are not valid characters in DNS hostnames.
```

Comment:
@milas @aevesdocker this probably seems micro-nitpicky of me, but: underscores _are_ valid in DNS hostnames (unless there have been recent RFC updates I'm unaware of that alter that?).

Note: I'm not challenging Docker's choice to migrate to use hyphens instead -- that seems fine.  But I think it'd be worth ensuring that the documentation is accurate.

Reply:
_I no longer work at Docker, so I'm commenting purely in a personal capacity as an OSS developer_

I don't think this warrants a change on its own: quick search supports that underscore is **not** widely regarded as acceptable in DNS names. The separator change was made for better compatibility with this real world expectation.

Reply:
To provide some missing background: while preparing to migrate a V1 `docker-compose.yml` file using the migration guide, I found some service names containing mid-name underscores (`foo_bar.invalid`) and began wondering about any network-level implications of renaming those.  The statement that underscores aren't valid in DNS names temporarily made me skip considering those implications (thinking: if they're not valid, they can't possibly be used in any connection-strings), until I remembered and checked later to find that they are valid (perhaps not recommendable, sure, but nonetheless valid).

We can probably chalk it up to my misunderstanding and hastiness.  In theory I'd like to reduce the possibility for other people to follow that same reasoning detour, and would happily offer a pull request, but I'm also OK if an edit for this is not considered worthwhile.